src/common/Gateway.sol

```diff
@@ -283,52 +283,55 @@ contract Gateway is Auth, Recoverable, IGateway {
 
             TransientBytesLib.append(batchSlot, message);
         } else {
-            _send(centrifugeId, poolId, message);
+            _send(centrifugeId, poolId, message, gasService.gasLimit(centrifugeId, message));
             _refundTransaction();
         }
     }
 
-    function _send(uint16 centrifugeId, PoolId poolId, bytes memory batch) internal returns (bool succeeded) {
+    function _send(uint16 centrifugeId, PoolId poolId, bytes memory batch, uint128 batchGasLimit)
+        internal
+        returns (bool succeeded)
+    {
         IAdapter[] memory adapters_ = adapters[centrifugeId];
         require(adapters[centrifugeId].length != 0, EmptyAdapterSet());
 
         SendData memory data = SendData({
```

Comment:
I wonder if the struct might not be necessary anymore, with your `{ }` trick and the removal of the gas limit?